File: src/algorithms/strings/caesar-cipher-encryptor/solution1.py
# O(n) time | O(n) space
def caesarCipherEncryptor(string, key):
	newLetters = []
	newKey = key % 26
	for letter in string:
		newLetters.append(getNewLetter(letter, newKey))
	return "".join(newLetters)

def getNewLetter(letter, key):
	newLetterCode = ord(letter) + key
	return chr(newLetterCode) if newLetterCode <= 122 else chr(96 + newLetterCode % 122)

File: src/algorithms/strings/caesar-cipher-encryptor/solution1.ts
// O(n) time | O(n) space
export function caesarCipherEncryptor(string: string, key: number) {
  const newLetters: string[] = [];
  const newKey = key % 26;
  for (const letter of string) {
    newLetters.push(getNewLetter(letter, newKey));
  }
  return newLetters.join("");
}

function getNewLetter(letter: string, key: number) {
  const newLetterCode = letter.charCodeAt(0) + key;
  return newLetterCode <= 122
    ? String.fromCharCode(newLetterCode)
    : String.fromCharCode(96 + (newLetterCode % 122));
}
